src/utils/object.ts

```diff
@@ -216,6 +216,56 @@ export function camelizeObjectKeys(obj: { [key: string]: any }) {
   return newObj;
 }
 
+/**
+ * camelizeKeys recursively turns object keys camelCased, works on
+ *   arrays of objects, and object-key's array values, which may
+ *   contain column names
+ * This does not replace camelizedObjectKeys because of its
+ *   aggressive (recursive) nature, which is a little more greedy
+ *
+ * @param arg <Array|object|string>
+ * @returns An object with camelized keys or camelized string
+ */
+export function camelizeKeys(arg: any) {
+  let response: any;
+
+  if (typeof arg === 'string') {
+    return camelize(arg);
+  } else if (Array.isArray(arg)) {
+    response = [];
+
+    for (const [i, element] of arg.entries()) {
+      if (typeof element === 'string') {
+        response[i] = camelize(element);
+      } else { // array/object/etc
+        response[i] = camelizeKeys(element);
+      }
+    }
+  } else if (typeof arg === 'object') {
```

Comment:
use `isPlainObject` instead. `null` has a typeof `object` and this method should only camelize plain objects (i.e. `{}`, `Object.create(null)`, but not `new Date()`)

Reply:
Good point.

Some questions:
1. is `null` ever passed in as a value?
2. is it ever a return value or set as the value of a key?  
 
I would think the database response converts nulls to string form (i.e. `'NULL'`).  If that's true, someone would have to explicitly pass in `null` for camelizeKeys to have in issue -- in which case they get what they deserve 😆

This seems to be overkill for what the function is doing.  I suspect you're future planning on how it could be used with values that are not strings or collections (arrays/objects); for instance, dates objects (or other objects) aren't returned by foreign key lookup queries.

That might be solving a future use case, which is beyond the scope of what the goal here was.

Reply:
> I would think the database response converts nulls to string form (i.e. 'NULL')

I don't see why the database would convert a null value to `'null'`. We would have no way of differentiating these two values